### DISCUSSION TITLE
Plotting skin diffuse reflectance spectra using mcmatlab

### DISCUSSION THREAD

**Discussion by leoshmu** [ankrh/MCmatlab#20] [Q&A] [answered]
Hello all, 
We have been using MCMatlab for determining the thermal response of tissue to laser stimulation and it works great. We recently wanted to apply MCMatlab to extract diffuse reflectance spectra, similar to what was done in FIgure 3b of this paper https://www.sciencedirect.com/science/article/pii/S101113440600251X. Another example in the literature is this paper where they don't use MCMatlab but do use monte carlo to predict a skin relfectance spectrum: https://core.ac.uk/download/pdf/140734.pdf. It seems like MCMatlab could be used to recreate their analysis. Is example 23 of the github repository one that could be applied? It seems like you just need to run a simulation at wavelengths from 400-700 for example and take a ratio of photons launched to photons reflected to be able to plot diffuse reflectance? Or is there a different example that is better to use or that shows the diffuse reflectance calculation? Thanks in advance!

**Comment by ankrh** [comment, marked as answer]
Thanks for your use of MCmatlab!

Yes, MCmatlab should be perfect for your simulations and example 23 would be the best starting point. Assuming you have no specular reflection (`model.MC.matchedInterfaces = true`), the expression you can use to get the diffuse reflectance is

`model.G.dx*model.G.dy*sum(model.MC.normalizedIrradiance_zneg,'all')`

model.MC.normalizedIrradiance_zneg is already normalized to the number of launched photons so you don't need to worry about that. See also example 20, where specular reflection is subtracted.

If you're interested in other results such as radial reflectance or transmittance you can take inspiration from example 24.

**Reply by leoshmu** [reply]
Thanks so much, this is exactly what we're doing, excited to share more once we complete our analysis
